tap_salesforce/salesforce/bulk.py

```diff
@@ -190,6 +202,17 @@ def _add_batch(self, catalog_entry, job_id, start_date, order_by_clause=True):
 
         return batch['batchInfo']['id']
 
+    def _complete_batch(self, state, tap_stream_id, batch_id):
```

Comment:
I added this method b/c when a pk chunked job was complete, the bookmark still had the `JobID` and `BatchIDs` properties in it. The JobID needs to be cleared from the state, otherwise on the next syncing interval, it would incorrectly think it was resuming a failed batch (but the `BatchIDs` array was empty).

See:
https://github.com/MeltanoLabs/tap-salesforce/blob/d802f68929d9621a72a9639603b422ea90490520/tap_salesforce/__init__.py#L453